### ISSUE TITLE
Setting maxReplicas (for hpa) results in wrong scaleTargetRef.kind when mode is statefulset 

### ISSUE THREAD

**Issue by jsirianni** [open-telemetry/opentelemetry-operator#922]
I have a config like this

```yaml
apiVersion: opentelemetry.io/v1alpha1
kind: OpenTelemetryCollector
metadata:
  name: gcp-gateway
  namespace: default
  labels:
    app: gcp-gateway
spec:
  mode: statefulset
...
  replicas: 2
  maxReplicas: 5
```

The HPA looks like this (points to a deployment):
```
observiq-gcp-gateway-collector   Deployment/observiq-gcp-gateway-collector   <unknown>/90%   2         5         0          35s
```

The HPA is correct except for its target. 

After re-thinking my approach, I can use a Deployment for my collector, but I think this is still a bug. The operator should create the correct HPA.

**Comment by pavolloffay** [comment]
cc) @kevinearls

**Comment by pavolloffay** [comment]
@kevinearls was this resolved by #984 ?

**Comment by kevinearls** [comment]
@pavolloffay Yes
